ui/pages/_app.js

```diff
@@ -681,105 +684,109 @@ class MesheryApp extends App {
-                    <nav
-                      className={isDrawerCollapsed ? classes.drawerCollapsed : classes.drawer}
-                      data-test="navigation"
-                      id="left-navigation-bar"
-                      style={{ height: '100%', overflow: 'visible' }}
-                    >
-                      <Hidden smUp implementation="js">
-                        <Navigator
-                          variant="temporary"
-                          open={this.state.mobileOpen}
-                          onClose={this.handleDrawerToggle}
-                          onCollapseDrawer={(open = null) => this.handleCollapseDrawer(open)}
-                          isDrawerCollapsed={isDrawerCollapsed}
-                          updateExtensionType={this.updateExtensionType}
-                        />
-                      </Hidden>
-                      <Hidden xsDown implementation="css">
-                        <Navigator
-                          onCollapseDrawer={(open = null) => this.handleCollapseDrawer(open)}
-                          isDrawerCollapsed={isDrawerCollapsed}
-                          updateExtensionType={this.updateExtensionType}
-                        />
-                      </Hidden>
-                    </nav>
-                  )}
-                  <div className={classes.appContent}>
-                    <SnackbarProvider
-                      anchorOrigin={{
-                        vertical: 'bottom',
-                        horizontal: 'right',
-                      }}
-                      iconVariant={{
-                        success: <CheckCircle style={{ marginRight: '0.5rem' }} />,
-                        error: <Error style={{ marginRight: '0.5rem' }} />,
-                        warning: <Warning style={{ marginRight: '0.5rem' }} />,
-                        info: <Info style={{ marginRight: '0.5rem' }} />,
-                      }}
-                      Components={{
-                        info: ThemeResponsiveSnackbar,
-                        success: ThemeResponsiveSnackbar,
-                        error: ThemeResponsiveSnackbar,
-                        warning: ThemeResponsiveSnackbar,
-                        loading: ThemeResponsiveSnackbar,
-                      }}
-                      maxSnack={10}
-                    >
-                      <NotificationCenterProvider>
-                        <MesheryProgressBar />
-                        {!this.state.isFullScreenMode && (
-                          <Header
-                            onDrawerToggle={this.handleDrawerToggle}
-                            onDrawerCollapse={isDrawerCollapsed}
-                            contexts={this.state.k8sContexts}
-                            activeContexts={this.state.activeK8sContexts}
-                            setActiveContexts={this.setActiveContexts}
-                            searchContexts={this.searchContexts}
+            {this.state.isLoading ? (
```

Comment:
please dont use a terninary operator here . it makes the whole block messy and hard to read . create a Loading component that take children and isLoading as props and renderes accordingly